src/components/DataTable/DataTable.tsx

```diff
@@ -159,129 +176,141 @@ export function DataTable<T>({
    * header, search field, and actions, and preserve accessibility.
    */
   return (
-    <div className={componentClassName} {...other}>
-      {(caption || subcaption || onSearchChange || actions) && (
-        <div className={styles['data-table__caption-container']}>
-          {(caption || subcaption) && (
-            <div className={styles['data-table__caption-text']}>
-              {caption && (
-                <div
-                  aria-hidden="true"
-                  className={styles['data-table__caption']}
-                >
-                  {caption}
-                </div>
-              )}
-              {subcaption && (
-                // TODO: Warn when only using subcaption
-                <div
-                  aria-hidden="true"
-                  className={styles['data-table__subcaption']}
-                >
-                  {subcaption}
-                </div>
-              )}
-            </div>
-          )}
-          {onSearchChange && (
-            <div className={styles['data-table__search']}>
-              <DataTableSearch />
-            </div>
-          )}
-          {actions && (
-            <div className={styles['data-table__actions']}>
-              <DataTableActions>{actions}</DataTableActions>
-            </div>
-          )}
-        </div>
-      )}
-      {/* Provide an escape hatch for specifying all aspects of a table using `children` and Sub-components directly */}
-      {children ?? (
-        <DataTableTable
-          className={tableClassName}
-          rowStyle={rowStyle}
-          size={size}
-          tableStyle={tableStyle}
-        >
-          {(caption || subcaption) && (
-            <caption className={styles['data-table__aria-caption']}>
-              {`${caption}${subcaption ? ': ' + subcaption : ''}`}
-            </caption>
-          )}
-          <DataTableHeader>
-            {table?.getHeaderGroups().map((headerGroup) => (
-              <DataTableRow key={headerGroup.id}>
-                {headerGroup.headers.map((header) => (
-                  <th
-                    className={styles['data-table__header-cell-container']}
-                    colSpan={header.colSpan}
-                    key={header.id}
-                    style={{
-                      width: `${header.getSize()}px`,
-                    }}
+    <DataTableContext.Provider value={{ size }}>
```

Comment:
Most of this diff is whitespace changes. Use the ⚙️ above to update the view to hide whitespace changes.